binder/src/main/java/io/grpc/binder/internal/BinderClientTransport.java

```diff
@@ -316,14 +316,21 @@ void notifyTerminated() {
       readyTimeoutFuture.cancel(false);
       readyTimeoutFuture = null;
     }
-    if (preAuthResultFuture != null) {
-      preAuthResultFuture.cancel(false); // No effect if already complete.
-    }
-    if (authResultFuture != null) {
-      authResultFuture.cancel(false); // No effect if already complete.
-    }
+
+    ListenableFuture<Status> preAuthFuture = preAuthResultFuture;
```

Comment:
I find these new local variable names confusing because they are similar to the member names yet different in a way that doesn't give any hint to the purpose (effectively final)

Instead, how about a new method like:

// Enqueues a future for cancellation later, on another thread.
// Useful when the caller wants to cancel while holding locks but the future is visible to
// user code which might have added listeners to run on directExecutor().
private cancelAsync(@Nullable ListenableFuture<?> future) {
  ...
}

?

Reply:
I think it is a good idea. Thank you!

Code review feedback has been applied.

Reply:
Thanks. On second thought, how about calling this `maybeCancelAsync()` because of the `!isDone()` check?

Reply:
OK point well taken about the word "maybe".  Good method names convey their intended purpose and guarantees and "maybe" leaves the reader wondering what those are. `cancelAsyncIfIsNotDone` is better I guess but feels like it wastes readers attention on implementation details. How about `cancelAsyncIfNeeded`, since there's actually a couple reasons cancellation might not be needed, but the reader doesn't really need to know them. The original 'cancelAsync' would be fine too.